forui/lib/src/widgets/accordion.dart

```diff
@@ -0,0 +1,525 @@
+  /// * Throws [AssertionError] if [max] < 0.
+  /// * Throws [AssertionError] if [min] > [max].
+  FAccordionController({
+    int min = 0,
+    int? max,
+    Set<int>? values,
+    this.duration = const Duration(milliseconds: 500),
+  })  : _min = min,
+        _max = max,
+        _controllers = [],
+        _values = values ?? {},
+        assert(min >= 0, 'The min value must be greater than or equal to 0.'),
+        assert(max == null || max >= 0, 'The max value must be greater than or equal to 0.'),
+        assert(max == null || min <= max, 'The max value must be greater than or equal to the min value.');
+
+  /// Adds an item to the accordion.
+  void addItem(int index, AnimationController controller, Animation expand) {
+    _controllers.add((controller, expand));
+    if (controller.value == 1.0) {
+      _values.add(index);
+    }
+  }
+
+  /// Removes an item from the accordion.
+  void removeItem(int index) {
+    _controllers.removeAt(index);
+    _values.remove(index);
+  }
+
+  /// Convenience method for toggling the current expanded status.
+  ///
+  /// This method should typically not be called while the widget tree is being rebuilt.
+  Future<void> toggle(int index) async {
+    if (_controllers[index].$2.value == 100) {
+      if (_values.length <= _min) {
+        return;
+      }
+      _values.remove(index);
+      await collapse(index);
+    } else {
+      if (_max != null && _values.length >= _max) {
+        return;
+      }
+      _values.add(index);
+      await expand(index);
+    }
+    notifyListeners();
+  }
+
+  /// Shows the content in the accordion.
+  ///
+  /// This method should typically not be called while the widget tree is being rebuilt.
+  Future<void> expand(int index) async {
+    final controller = _controllers[index].$1;
+    await controller.forward();
+    notifyListeners();
+  }
+
+  /// Hides the content in the accordion.
+  ///
+  /// This method should typically not be called while the widget tree is being rebuilt.
+  Future<void> collapse(int index) async {
+    final controller = _controllers[index].$1;
+    await controller.reverse();
+    notifyListeners();
+  }
+}
+
+/// An [FAccordionController] that allows one section to be expanded at a time.
+class FRadioAccordionController extends FAccordionController {
+  /// Creates a [FRadioAccordionController].
+  FRadioAccordionController({
+    int? value,
+    super.duration,
+    super.min,
+    int super.max = 1,
+  }) : super(values: value == null ? {} : {value});
+
+  @override
+  Future<void> toggle(int index) async {
+    final toggle = <Future>[];
+    if (!(_values.length <= _min)) {
+      if (!_values.contains(index) && _values.length >= _max!) {
+        toggle.add(collapse(_values.first));
+        _values.remove(_values.first);
+      }
+    }
+    toggle.add(super.toggle(index));
+    await Future.wait(toggle);
+  }
+}
+
+/// A vertically stacked set of interactive headings that each reveal a section of content.
```

Comment:
We're missing a list of links to important/relevant stuff, i.e. docs, controllers, style.